src/main/kotlin/uk/gov/justice/digital/hmpps/prisonertonomisupdate/personalrelationships/profiledetails/ContactPersonReconciliationService.kt

```diff
@@ -0,0 +1,80 @@
+import uk.gov.justice.digital.hmpps.prisonertonomisupdate.personalrelationships.profiledetails.ContactPersonProfileType.CHILD
+import uk.gov.justice.digital.hmpps.prisonertonomisupdate.personalrelationships.profiledetails.ContactPersonProfileType.MARITAL
+import uk.gov.justice.digital.hmpps.prisonertonomisupdate.profiledetails.ProfileDetailsNomisApiService
+import uk.gov.justice.digital.hmpps.prisonertonomisupdate.services.doApiCallWithRetries
+
+@Service
+class ContactPersonReconciliationService(
+  @Autowired private val nomisApi: ProfileDetailsNomisApiService,
+  @Autowired private val dpsApi: ContactPersonProfileDetailsDpsApiService,
+  @Autowired private val telemetryClient: TelemetryClient,
+) {
+
+  suspend fun checkPrisoner(prisonerNumber: String): String? {
+    val apiResponses = withContext(Dispatchers.Unconfined) {
+      ApiResponses(
+        async { doApiCallWithRetries { nomisApi.getProfileDetails(prisonerNumber, ContactPersonProfileType.all()) } },
+        async { doApiCallWithRetries { dpsApi.getDomesticStatus(prisonerNumber) } },
+        async { doApiCallWithRetries { dpsApi.getNumberOfChildren(prisonerNumber) } },
+      )
+    }
+
+    return findDifferences(apiResponses)
+      ?.let { differences ->
+        prisonerNumber
+          .also {
+            telemetryClient.trackEvent(
+              "contact-person-profile-details-reconciliation-prisoner-failed",
+              mapOf("offenderNo" to it, "differences" to differences.joinToString()),
+              null,
+            )
+          }
+      }
+  }
+
+  private fun findDifferences(apiResponses: ApiResponses): List<String>? {
+    val differences = mutableListOf<String>()
+
+    val nomisDomesticStatus = apiResponses.nomisProfileDetails?.findDomesticStatusCode(MARITAL.name)
+    val dpsDomesticStatus = apiResponses.dpsDomesticStatus?.domesticStatusCode
+    if (nomisDomesticStatus != dpsDomesticStatus) {
+      differences += MARITAL.identifier
+    }
+
+    val nomisNumberOfChildren = apiResponses.nomisProfileDetails?.findDomesticStatusCode(CHILD.name)
+    val dpsNumberOfChildren = apiResponses.dpsNumberOfChildren?.numberOfChildren
+    if (nomisNumberOfChildren != dpsNumberOfChildren) {
+      differences += CHILD.identifier
+    }
+
+    return differences.takeIf { it.isNotEmpty() }
+  }
+
+  private data class ApiResponses(
+    val nomisProfileDetails: PrisonerProfileDetailsResponse?,
+    val dpsDomesticStatus: SyncPrisonerDomesticStatusResponse?,
+    val dpsNumberOfChildren: SyncPrisonerNumberOfChildrenResponse?,
+  ) {
+    companion object {
+      suspend operator fun invoke(
+        nomisProfileDetails: Deferred<PrisonerProfileDetailsResponse?>,
+        dpsDomesticStatus: Deferred<SyncPrisonerDomesticStatusResponse?>,
+        dpsNumberOfChildren: Deferred<SyncPrisonerNumberOfChildrenResponse?>,
+      ) = ApiResponses(nomisProfileDetails.await(), dpsDomesticStatus.await(), dpsNumberOfChildren.await())
+    }
```

Comment:
You can't have a suspend constructor - but you can have a suspend invoke operator!